_source/_posts/2019-10-25-intro-android-gradle.md

```diff
@@ -0,0 +1,392 @@
+Image from [Android Documentation](https://developer.android.com/studio/build/index.html)
+
+You can configure the Gradle project by modifying the `/build.gradle` file at the root of the project. These changes will affect the whole project, including all modules.
+
+Each module can be configured in each module-specific *build.gradle* file. In our case, for instance, we can modify the `/app/build.gradle` file to  change how Gradle handles the *app* module.
+
+## Essential Configuration Options for Gradle in Android
+
+Android users currently run a wide variety of [Android versions](https://developer.android.com/about/dashboards).
+To allow apps built for newer Android versions to run in older Android versions as well, developers must include many "compatibility layers" in apps.
+
+For instance, if an app built for Android 9 runs on an  Android 8 device, this device will run the code compiled for Android 9, plus a compatibility layer that translated some of the Android 9 functionality into something Android 8 can understand. As a result, he app will be bigger, it will run more slowly on Android 8 devices, and developers may have to add additional code to support Android 8.
+
+Multiple Gradle configurations allow developers to support as many Android versions as they judge appropriate, all of them configured at a module level. For instance, in our case we have:
+
+```gradle
+android {
+    compileSdkVersion 29
+    defaultConfig {
+        ...
+        minSdkVersion 24
+        targetSdkVersion 29
+        ...
+    }
+    ...
+}
+```
+
+This configuration determines that our app will be compiled using the 29th version of Android (Android 10) and should be compatible with any device running any version of Android between the 24th (Android 7.0) and the 29th. This means that our app will contain multiple compatibility layers to make Android 10 compatible with Android 7.0.
+
+### Reference Repositories
+
+Any time we reference external resources like dependencies in our project, Gradle must validate their existence and fetch them into our app. You also must configure Gradle with a list of  locations for these repositories, usually defined at a project-level.
+
+In our example, we can have a look at `/build.gradle`:
+
+```groovy
+buildscript {
+    ext.kotlin_version = '1.3.41'
+    repositories {
+        google()
+        jcenter()
+
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.4.2'
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        // NOTE: Do not place your application dependencies here; they belong
+        // in the individual module build.gradle files
+    }
+}
+
+allprojects {
+    repositories {
+        google()
+        jcenter()
+    }
+}
+```
+
+The *repositories{}* block in *buildscript {}* defines which repositories to check to satisfy the dependencies defined in the *buildscript { dependencies{} }* block. As we will soon learn, these dependencies are Gradle plugins.
```

Comment:
nit: this looks funny to me `repositories{}`and `buildscript { dependencies{} }`, i would expect the latter to be described as `buildscript.dependencies`